OBSDOCS-1294 Observability UI plugins overview (#81402)
openshift/openshift-docs: observability/cluster_observability_operator/ui_plugins/observability-ui-plugins-overview.adoc

:_mod-docs-content-type: ASSEMBLY
[id="observability-ui-plugins-overview"]
= Observability UI plugins overview
include::_attributes/common-attributes.adoc[]
:context: observability-ui-plugins-overview

toc::[]

:FeatureName: The Cluster Observability Operator
include::snippets/technology-preview.adoc[leveloffset=+2]

> The {coo-first} can install and manage UI plugins to enhance the observability capabilities of the OpenShift web console. The plugins extend the default functionality, providing new UI features for monitoring, troubleshooting, distributed tracing, and cluster logging.

Author: This is not a product name, just a generic term

== Dashboards

The dashboard UI plugin supports enhanced dashboards in the OpenShift web console at *Observe* -> *Dashboards* . You can add other Prometheus datasources from the cluster to the default dashboards, in addition to the in-cluster datasource. This results in a unified observability experience across different data sources.

== Troubleshooting

The troubleshooting panel UI plugin provides observability signal correlation, powered by the open source Korrel8r project. With the troubleshooting panel available under *Observe* -> *Alerting* , you can easily correlate metrics, logs, alerts, netflows, and additional observability signals and resources, across different data stores.
The output of Korrel8r is displayed in the form of an interactive node graph. When you click on a node, you are automatically redirected to the corresponding web console page with the specific information for that node, for example, metric, log, pod, etc. 

== Distributed tracing

> The distributed tracing UI plugin adds tracing-related features to the OpenShift web console under *Observe* -> *Traces*. You can follow requests through the front end and into the backend of microservices, helping you identify code errors and performance bottlenecks in distributed systems. Select a supported Tempo insance running the cluster, and set a time range and query, to view the trace data. 

Author: This is not a product name, just a generic term

> The distributed tracing UI plugin adds tracing-related features to the OpenShift web console under *Observe* -> *Traces*. You can follow requests through the front end and into the backend of microservices, helping you identify code errors and performance bottlenecks in distributed systems. Select a supported Tempo insance running the cluster, and set a time range and query, to view the trace data. 

Author: Changed to "Select a supported `TempoStack` or `TempoMonolithic` instance running in the cluster

:_mod-docs-content-type: ASSEMBLY
[id="observability-ui-plugins-overview"]
= Observability UI plugins overview
include::_attributes/common-attributes.adoc[]
:context: observability-ui-plugins-overview

toc::[]

> :FeatureName: The Cluster Observability Operator

Reviewer: Suggest adding an attribute for the Operator name if there isn't one already

Author: coo-full is there already - will use it here

Author: done

include::snippets/technology-preview.adoc[leveloffset=+2]

> The {coo-first} can install and manage UI plugins to enhance the observability capabilities of the {product-title} web console.

Reviewer: ```suggestion
You can use the {coo-first} to install and manage UI plugins to enhance the observability capabilities of the {product-title} web console.
```
Suggest revising to make [active](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice) and avoid [anthropomorphism](https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism)

Author: done

> The plugins extend the default functionality, providing new UI features for monitoring, troubleshooting, distributed tracing, and cluster logging.

Author: This is not a product name, just a generic term

== Dashboards

The dashboard UI plugin supports enhanced dashboards in the {product-title} web console at *Observe* -> *Dashboards*.

> You can add other Prometheus datasources from the cluster to the default dashboards, in addition to the in-cluster datasource.
> This results in a unified observability experience across different data sources.

Reviewer: ```suggestion
You can add other Prometheus data sources from the cluster to the default dashboards, in addition to the in-cluster data source.
This results in a unified observability experience across different data sources.
```

Both [IBMSG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#data-source) and [Prometheus docs](https://grafana.com/docs/grafana/latest/datasources/prometheus/) seem to use "data source".

Author: done

== Troubleshooting

The troubleshooting panel UI plugin provides observability signal correlation, powered by the open source Korrel8r project.

> With the troubleshooting panel available under *Observe* -> *Alerting* , you can easily correlate metrics, logs, alerts, netflows, and additional observability signals and resources, across different data stores.

Reviewer: ```suggestion
You can use the troubleshooting panel available from the *Observe* -> *Alerting* page to easily correlate metrics, logs, alerts, netflows, and additional observability signals and resources across different data stores.
```

- Suggestions for active voice / minimalism and readability.
- Avoid use of "under" preposition for accessibility: https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#location-of-interface-elements (not sure if **Alerting** is a page, menu, etc, so you may need to tweak this suggestion to fit)

Author: done

> The output of Korrel8r is displayed in the form of an interactive node graph. When you click on a node, you are automatically redirected to the corresponding web console page with the specific information for that node, for example, metric, log, pod, etc.

Reviewer: ```suggestion
The output of Korrel8r is displayed as an interactive node graph. When you click on a node, you are directed to the corresponding web console page with the specific information for that node.
```
Suggested editing for minimalism, I think the list of examples here might not be necessary. If including it, "etc" must not be used per https://www.ibm.com/docs/en/ibm-style?topic=word-usage#etc

Author: I will keep the list as it was added by an engineer but remove etc

== Distributed tracing

> The distributed tracing UI plugin adds tracing-related features to the OpenShift web console under *Observe* -> *Traces*.

Author: This is not a product name, just a generic term

> The distributed tracing UI plugin adds tracing-related features to the {product-title} web console under *Observe* -> *Traces*.

Reviewer: - Mixed usage of "{product-title} web console" and just "web console" throughout the doc - I'd choose one and keep it consistent. Maybe use "{product-title} web console" for the first instance and just "web console" for all subsequent instances?
- Same comment as previous about avoiding use of "under" or other prepositions like this for accessibility

Author: done

You can follow requests through the front end and into the backend of microservices, helping you identify code errors and performance bottlenecks in distributed systems.

> Select a supported `TempoStack` or `TempoMonolithic` instance running in the cluster, and set a time range and query, to view the trace data.

Reviewer: ```suggestion
You can select a supported `TempoStack` object or `TempoMonolithic` object running in the cluster and set a time range and query to view the trace data.
```
Make more active, also I don't think the commas here are necessary

Author: I think the instruction "Select" is pretty active so will leave it - also , eng have used instance explicitly here, so I will keep that. Removing commas done

> == CLuster logging

Reviewer: ```suggestion
["observability-ui-plugins-overview-logging_{context}"]
== Cluster logging
```

Typo in heading, also all the H2 (`==`) headings in this assembly need to have anchor IDs added. I've provided an example. More info: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assembly-subsections-level-1-and-lower-headings

Author: done

> The logging UI plugin surfaces logging data in the {product-title} web console under *Observe* -> *Logs*. 

Reviewer: - Mixed usage of "{product-title} web console" and just "web console" throughout the doc - I'd choose one and keep it consistent. Maybe use "{product-title} web console" for the first instance and just "web console" for all subsequent instances?
- Same comment as previous about avoiding use of "under" or other prepositions like this for accessibility

Author: Done

> You can specify filters, queries, time ranges and refresh rates, with the results displayed a list of collapsed logs, which can then be expanded to show more detailed information for each log.

Reviewer: ```suggestion
You can specify filters, queries, time ranges, and refresh rates. Results are displayed a list of collapsed logs, which can then be expanded to show more detailed information for each log.
```
Suggested edits for readability

Author: Done

:_mod-docs-content-type: ASSEMBLY
[id="observability-ui-plugins-overview"]
= Observability UI plugins overview
include::_attributes/common-attributes.adoc[]
:context: observability-ui-plugins-overview

toc::[]

:FeatureName: The {coo-full}
include::snippets/technology-preview.adoc[leveloffset=+2]

You can use the {coo-first} to install and manage UI plugins to enhance the observability capabilities of the {product-title} web console.
The plugins extend the default functionality, providing new UI features for monitoring, troubleshooting, distributed tracing, and cluster logging.

["dashboards_{context}"]
== Dashboards

The dashboard UI plugin supports enhanced dashboards in the {product-title} web console at *Observe* -> *Dashboards*.
You can add other Prometheus data sources from the cluster to the default dashboards, in addition to the in-cluster data source.
This results in a unified observability experience across different data sources.

["troubleshooting_{context}"]
== Troubleshooting

The troubleshooting panel UI plugin provides observability signal correlation, powered by the open source Korrel8r project.

> You can use the troubleshooting panel available from the *Observe* -> *Alerting* page to easily correlate metrics, logs, alerts, netflows, and additional observability signals and resources, across different data stores.

Reviewer 2: @gabriel-rh The troubleshooting panel can be also triggered from the masthead application launcher, as seen in this PR: https://github.com/openshift/troubleshooting-panel-console-plugin/pull/67/files. This is enabled form 4.17. Would you think this is something we can add in this paragraph?

Reviewer 2: I think this is already included here: https://github.com/openshift/openshift-docs/pull/81463/files#diff-e8d2db0b08d6c6517eb6812f32c2a129172a3511808372eda637b1046931505bR13

Author: Added here as well

The output of Korrel8r is displayed as an interactive node graph. When you click on a node, you are automatically redirected to the corresponding web console page with the specific information for that node, for example, metric, log, or pod.

["distributed-tracing_{context}"]
== Distributed tracing

The distributed tracing UI plugin adds tracing-related features to the web console on the *Observe* -> *Traces* page.
You can follow requests through the front end and into the backend of microservices, helping you identify code errors and performance bottlenecks in distributed systems.

> You can select a supported `TempoStack` or `TempoMonolithic` instance running in the cluster and set a time range and query to view the trace data.

Reviewer 2: We only support `TempoStack` with multi-tenancy enabled, other types like `TempoMonolithic` or `TempoStacks` without multi-tenancy are not shown

Author: Done
